inst/sql/sql_server/concept_plausible_unit_concept_ids.sql

```diff
@@ -39,8 +39,11 @@ FROM
     		AND c.cohort_definition_id = @cohortDefinitionId
     	}
 		WHERE m.@cdmFieldName = @conceptId
-		  	AND COALESCE (m.unit_concept_id, -1) NOT IN (@plausibleUnitConceptIds) -- '-1' stands for the cases when unit_concept_id is null
-		  	AND m.value_as_number IS NOT NULL 
+		  	AND /* '-1' stands for the cases when the only plausible unit_concept_id is no unit; 0 prevents flagging rows with a unit_concept_id of 0, which are checked in standardConceptRecordCompleteness */
+				CASE 
+					WHEN '@plausibleUnitConceptIds' = '-1' THEN COALESCE(m.unit_concept_id, -1) NOT IN (@plausibleUnitConceptIds) 
+					ELSE m.unit_concept_id NOT IN (@plausibleUnitConceptIds, 0) 
+				END
```

Comment:
We can simplify this to "fails if plausible -1 but unit not null, or unit not in plausible units".
Also, instead of letting 0 always pass, we should simply ignore it for the check (see next comment).
```suggestion
 (
 (@plausibleUnitConceptIds = -1 AND m.unit_concept_id IS NOT NULL)
 OR m.unit_concept_id NOT IN (@plausibleUnitConceptIds) 
 )
```